compiler/vm/vmcompilerserdes.nim

```diff
@@ -86,17 +86,26 @@ proc deserializeRef*(c: TCtx, slot: HeapSlotHandle, vt: PVmType; f, con: PType,
       # pass the unskipped base type
       result = c.deserializeObject(src.byteView(), vt.targetType, f, conBase, info)
     else:
+      # result = c.config.newError(
+      #   wrongNode(base),
+      #   SemReport(kind: rvmUnsupportedNonNil, typ: con))
       result = c.config.newError(
         wrongNode(base),
-        SemReport(kind: rvmUnsupportedNonNil, typ: con))
+        PAstDiag(kind: adVmUnsupportedNonNil, unsupported: con))
 
   else:
     let e = r.takeErr()
     result =
       if e == dfcNil:
         newNodeIT(nkNilLit, info, f)
       else:
-        c.config.newError(wrongNode(f), e.toReport())
+        # c.config.newError(wrongNode(f), e.toReport())
+        c.config.newError(wrongNode(f), PAstDiag(
```

Comment:
Double check, we might be able to drop the `VmEvent`/`VmEventKind` related stuff for this.

Reply:
`vmEvtUnsupportedNonNil` can definitely be removed now.